contracts/sg-ics721/src/testing/integration_tests.rs

```diff
@@ -0,0 +1,1556 @@
+                                num_tokens: 1,
+                            })
+                            .unwrap(),
+                        ),
+                    },
+                    tokens: vec![Token {
+                        id: TokenId::new("1"),
+                        uri: Some("https://moonphase.is/image.svg".to_string()),
+                        data: None,
+                    }],
+                },
+            }),
+            &[],
+        )
+        .unwrap();
+
+    // Check entry added in CLASS_ID_TO_NFT_CONTRACT
+    let class_id_to_nft_contract = test.query_nft_contracts();
+    assert_eq!(class_id_to_nft_contract.len(), 1);
+    assert_eq!(class_id_to_nft_contract[0].0, "bad kids");
+
+    // This will only do a mint as the contract for the class ID has
+    // already been instantiated.
+    test.app
+        .execute_contract(
+            test.ics721.clone(),
+            test.ics721.clone(),
+            &ExecuteMsg::Callback(CallbackMsg::CreateVouchers {
+                receiver: "mr-t".to_string(),
+                create: VoucherCreation {
+                    class: Class {
+                        id: ClassId::new("bad kids"),
+                        uri: Some("https://moonphase.is".to_string()),
+                        // unlike above in 1st transfer, here on 2nd transfer no classdata is provided!
+                        // this won't affect collection since it's already instantiated
+                        data: None,
+                    },
+                    tokens: vec![Token {
+                        id: TokenId::new("2"),
+                        uri: Some("https://moonphase.is/image.svg".to_string()),
+                        data: None,
+                    }],
+                },
+            }),
+            &[],
+        )
+        .unwrap();
+
+    // Check no additional entry added in CLASS_ID_TO_NFT_CONTRACT
+    let class_id_to_nft_contract = test.query_nft_contracts();
+    assert_eq!(class_id_to_nft_contract.len(), 1);
+
+    // Get the address of the instantiated NFT.
+    let nft: Addr = test
+        .app
+        .wrap()
+        .query_wasm_smart(
+            test.ics721.clone(),
+            &QueryMsg::NftContract {
+                class_id: "bad kids".to_string(),
+            },
+        )
+        .unwrap();
+
+    // check collection info is properly set
+    let collection_info: CollectionInfoResponse = test
+        .app
+        .wrap()
+        .query_wasm_smart(nft.clone(), &Sg721QueryMsg::CollectionInfo {})
+        .unwrap();
+    let (_source_hrp, source_data, source_variant) = bech32::decode(OWNER_SOURCE_CHAIN).unwrap();
+    let target_owner = bech32::encode(TARGET_HRP, source_data, source_variant).unwrap();
+
+    assert_eq!(
+        collection_info,
+        CollectionInfoResponse {
+            // creator is set to owner as defined by ClassData in 1st transfer!
+            creator: target_owner,
```

Comment:
Though 2nd inbound has no class data, creator is defined - due to instantiation of collection during 1st transfer.